src/Libraries/Microsoft.Extensions.Http.Telemetry/Logging/Internal/Log.cs

```diff
@@ -40,22 +61,22 @@ public static void OutgoingRequestError(ILogger logger, LogRecord record, Except
 
             collector.AddRequestHeaders(record.RequestHeaders);
             collector.AddResponseHeaders(record.ResponseHeaders);
-            collector.Add(HttpClientLoggingDimensions.Host, record.Host);
             collector.Add(HttpClientLoggingDimensions.Method, record.Method);
+            collector.Add(HttpClientLoggingDimensions.Host, record.Host);
             collector.Add(HttpClientLoggingDimensions.Path, record.Path);
             collector.Add(HttpClientLoggingDimensions.Duration, record.Duration);
 
-            if (record.StatusCode is not null)
+            if (record.StatusCode.HasValue)
             {
-                collector.Add(HttpClientLoggingDimensions.StatusCode, record.StatusCode);
+                collector.Add(HttpClientLoggingDimensions.StatusCode, record.StatusCode.Value);
             }
 
-            if (!string.IsNullOrEmpty(record.RequestBody))
+            if (record.RequestBody is not null)
             {
                 collector.Add(HttpClientLoggingDimensions.RequestBody, record.RequestBody);
             }
 
-            if (!string.IsNullOrEmpty(record.ResponseBody))
+            if (record.ResponseBody is not null)
```

Comment:
Checking just for `null` will be much faster

Reply:
Trivially faster I think.

So what's the behavior when the body is empty?

Reply:
We'll log it as a separate dimension (but that would mean that its logging was enabled via options, but its content was empty). Previous logic was dropping it from being emitted either way.